README.md
# Orchestra Counter

This project is the standard Orchestra Counter Application found in Orchestra 7

## Getting Started

These instructions will help you get started running this project on your local machine for,
1. Development
    1. Orchestra Installed on your local machine
    2. Orchestra Installed on a remote machine
2. Creating the Development Build
3. Creating the Production Build
4. Creating the Utt files

## What is include in this project

1. Web App
2. Utts
3. Language properties file

## Prerequisites

Make sure you have installed 
1. Node 8 or above.
2. LiveReload on Chrome (Optional) - Required only for development
    https://chrome.google.com/webstore/detail/livereload/jnihajbhpnppcggbcgedagnkighmdlei

## Development

1. Clone the project
2. Run the following command to install all dependencies

```
npm install
```
3. Copy the properties file found in the lang folder into C:\qmatic\orchestra\system\conf\lang folder of your orchestra system.

### Orchestra Installed on your local machine

In the case you are developing with an Orchestra installed on your local machine.

To start development run the following command

```
npm start
```

If the application doesn't open in your chrome browser automatically, you can visit **localhost:1337** to see the application. (If you want to see any of the changes you make in the code auto update the application in the browser, make sure live reload is enabled on the browser.)

### Orchestra Installed on a remote machine

In the case you are developing with an Orchestra installed on a remote machine.

Create a config.gulp.json file in the root of the project and make sure to add the remote ip,port and host details in the structure shown below.

```
{
    "proxy": {
        "host": "10.2.2.210",
        "port": "8080",
        "protocol": "http"
    }
}
```

To start development run the following command

```
npm start
```

If the application doesn't open in your chrome browser automatically, you can visit **localhost:1337** to see the application. (If you want to see any of the changes you make in the code auto update the application in the browser, make sure live reload is enabled on the browser.)

## Creating the Development Build

To create a development build(war file/ properties file and utts), run the following command

```
npm run build-war-dev
```

You will find the files in the dist folder.

## Creating the Production Build

To create a production build(war file/ properties file and utts), run the following command

```
npm run build-war-prod
```

You will find the files in the dist folder.
**Note: This build is more optimized for production than the development build.**

## Creating the Utt files

If you have updated the utt source files(.xml, etc), use the following command to build all the utts.

```
npm run build-utts
```

You will find the new utt files in their respective folders, inside the utt folder.

## License

### Apache Font License
Copyright 2018 Qmatic

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.


release-notes/release-notes.md
<h1>Release notes Orchestra Web Counter 4.0.0</h1>

----------

<h2>Introduction</h2>

This document describes the new features, bug corrections, known issues and recommendations for Orchestra Web Counter 1.0.0. If you want to know about connector changes details or similar, this document is for you.

**Note:** Several of the remarks refer to a Jira number (Jira is Qmatic&#39;s internal registration system for bugs), or Pivotal Tracker (internal system for improvements and other issues).

<h2>Version 4.0.0.032</h2>

**Date: 28/02/2020**

**Build number: 032**

<h3>Stories</h3>

| **Id** | **Release notes** |
| --- | --- |
| **171250252** | **Update customer details of an unknown customer** |
| **169089830** | **General Counter- new to save newline characters to Counter notes field.** |

<h3>Bug fixes</h3>

| **Id** | **Release notes** |
| --- | --- |
| **170643047** | **Update customer information is not able to save in the workstation** |
| **170049536** | **Ticket list in queue cannot be sorted, neither ascending nor descending works** |

----------

<h2>Version 4.0.0.031</h2>

**Date: 05/11/2019**

**Build number: 031**

<h3>Stories</h3>

| **Id** | **Release notes** |
| --- | --- |
| **165239364** | **Transaction / Service Time** |
| **164825164** | **Presentation of visits (tickets) by work profile (GAP Spain)** |
| **164861500** | **Present additional information per visit in user pool/counter pool (GAP BNLX))** |
| **168215794** | **Arrow key navigation in lists** |

<h3>Bug fixes</h3>

| **Id** | **Release notes** |
| --- | --- |
| **169444525** | **Counter UI > User cannot see visits at the bottom of the visit list.** |

----------

<h2>Version 4.0.0.029</h2>

**Date: 30/08/2019**

**Build number: 029**

<h3>Stories</h3>

| **Id** | **Release notes** |
| --- | --- |
| **167084467** | **Indicate charachter limit to user** |
| **000000000** | **Added WCAG support** |

<h3>Bug fixes</h3>

| **Id** | **Release notes** |
| --- | --- |
| **167338530** | **GUI issue in create customer form** |
| **167338834** | **Create customer form validation issues** |
| **167776865** | **Visit card > After add a service to the visit, user will see the visit card behind the add service window.** |
| **167776866** | **When there is no Mark type is set from UTT, not to show 'Add a mark' link in the visit card.** |
| **167993307** | **Walk in customer > service search is not working.** |
| **168070913** | **Serve Page / Note field character limit exceed message** |
| **167421044** | **date of birth" field label text is not reading from screen reader (NVDA)** |
| **167452543** | **WCAG - Add customer > mandatory fields are getting red once user tab on it.** |
| **167479893** | **Add Marks > validation issues in no of marks field** |

----------

<h2>Version 4.0.0.027</h2>

**Date: 05/06/2019**

**Build number: 027**

<h3>Stories</h3>

| **Id** | **Release notes** |
| --- | --- |
| **166081841** | **WCAG - CP10 - Contrast between foreground and background** Change colors, added validation messages to customer forms |
| **166026887** | **WCAG - Tabbing NL20/NL30** Improve tabbing, change focus to popovers on initialization and destruction, change styling to indicate focus on custom tabable elements, group links in header |
| **166080611** | **WCAG - Bypass Blocks - NL50** Add anchor link shortcuts to card and action panel |
| **166030010** | **WCAG - SW60 - Move Focus ** Move focus to card top when pressing WALK-IN, Add service to visit, Add marks, Call next, Add Ds/Outcome |

----------

<h2>Version 4.0.0.022</h2>

**Date: 15/04/2019**

**Build number: 022**

<h3>Stories</h3>

| **Id** | **Release notes** |
| --- | --- |
| **164824646** | **Call any service in a multi service visit** Add serve buttons to multi services in visit card |
| **COUNTER-126** | **Setting to set several Mark Types for a counter and show the marks grouped in the Counter** Add dropdown to select mark type before selecting mark |
| **COUNTER-127** | **Remove users active session (GAP BNLX/UK)** Add option to force logout |
| **COUNTER-129** | **Transfer with delay for visit/multi-service visit (GAP BNLX)** Add option to transfer with delay from queue |
| **166081841** | **WCAG - CP10 - Contrast between foreground and background** Change colors, added validation messages to customer forms |
| **166026887** | **WCAG - Tabbing NL20/NL30** Improve tabbing, change focus to popovers on initialization and destruction, change styling to indicate focus on custom tabable elements, group links in header |
| **166080611** | **WCAG - Bypass Blocks - NL50** Add anchor link shortcuts to card and action panel |
| **166030010** | **WCAG - SW60 - Move Focus ** Move focus to card top when pressing WALK-IN, Add service to visit, Add marks, Call next, Add Ds/Outcome |

<h3>Bug fixes</h3>

| **Id** | **Release notes** |
| --- | --- |
| **COUNTER-134** | **Cannot call next visit** Removed loop checking visit state during visit in display queue, added checks for empty responses in queues and pools |
| **COUNTER-137** | **custom_1 field includes too many characters and makes statistics end up in failed events** Removed encoding when saving notes |
| **COUNTER-138** | **LED does not show called ticket after StoreNext call from WebServicepoint** UTT adjustments to REMOVE_USER_FROM_STORE_NEXT |
| **COUNTER-140** | **Hard to see Mark Types options on small screens** Adjustment of dropdown heights for marks dropdowns |
| **COUNTER-143** | **Need to click TWICE after search** Fixed event listener for custom search |
| **COUNTER-145** | **workstationTerminalMessages.properties > spelling mistakes in English phases** Corrected spelling |

----------

<h2>Version 4.0.0.021</h2>

**Date: 18/03/2019**

**Build number: 021**

<h3>Stories</h3>

| **Id** | **Release notes** |
| --- | --- |
| **164398417** | **counter story** Add cancel button to multi service modal |
| **163318753** | **Use system parameter for date** Use system parameter when displaying date |

gulpfile.js
const gulp = require('gulp');
const sass = require('gulp-sass');
const autoprefixer = require('gulp-autoprefixer');
const del = require('del');
const nunjucksRender = require('gulp-nunjucks-render');
const connect = require('gulp-connect');
const devServer = connect;
var proxy = require('http-proxy-middleware');
const zip = require('gulp-zip');
var sftp = require('gulp-sftp');
var cachebust = require('gulp-cache-bust');

var useref = require('gulp-useref');
var gulpif = require('gulp-if');
var uglify = require('gulp-uglify');
var minifyCss = require('gulp-clean-css');
var lineEndCorrector = require('gulp-line-ending-corrector');
var htmlmin = require('gulp-htmlmin');
var rename = require('gulp-rename');

var fs = require('fs');
var path = require('path');
var folders = require('gulp-folders-4x');
var ncmd = require('node-cmd');

var uttsPath = './utt';

// Remote Deployment Defaults
var remoteDeploymentDefaultPath = 'C:\\qmatic\\orchestra\\system\\custdeploy';
var remoteDeploymentDefaultLangPath =
  'C:\\qmatic\\orchestra\\system\\conf\\lang';
var remoteDeploymentDefaultHost = 'localhost';
var remoteDeploymentPlatform = 'windows';

// Configuration
// =============

require('events').EventEmitter.prototype._maxListeners = 100;

try {
  var config = require('./gulp.config.json');

  // Needed for dev server
  var targetOrchestraIp = config.proxy.host ? config.proxy.host : "localhost";
  var targetOrchestraPort = config.proxy.port ? config.proxy.port : "8080";
  var targetOrchestraProtocol = config.proxy.protocol ? config.proxy.protocol : "http";
  var targetOrchestraUrl = targetOrchestraProtocol + '://' + targetOrchestraIp + ':' + targetOrchestraPort;

  // Must be provided via config.gulp.json file
  var remoteDeployHost = config.remote_deploy.host
    ? config.remote_deploy.host
    : remoteDeploymentDefaultHost;
  var remoteDeployUsername = config.remote_deploy.username;
  var remoteDeployPassword = config.remote_deploy.password;

  // Artifactory Deployment (artifactory)
  var targetArtifactoryIp = config.artifactory.host
    ? config.artifactory.host
    : '';
  var targetArtifactoryPath = config.artifactory.path
    ? config.artifactory.path
    : '';
  var targetArtifactoryPort = config.artifactory.port
    ? config.artifactory.port
    : '80';
  var targetArtifactoryProtocol = config.artifactory.protocol
    ? config.artifactory.protocol
    : 'http';
  var targetArtifactoryUsername = config.artifactory.username;
  var targetArtifactoryPassword = config.artifactory.password;
  var targetArtifactoryUrl =
    targetArtifactoryProtocol +
    '://' +
    targetArtifactoryIp +
    ':' +
    targetArtifactoryPort;
} catch (ex) {
  // For those who don't provide an external configuration file, use the following default.
  // Assuming Orchestra is running on local machine
  if(targetOrchestraUrl === undefined) {
    var targetOrchestraUrl = "http://localhost:8080";
  }
  console.log('No gulp.config.json found. Assuming orchestra is running on local machine at port 8080');
}

// Tasks
// =====

function cleanBuild(done) {
  del.sync(['./dist']);
  done();
}

gulp.task('clean:build', cleanBuild);

function cleanWar(done) {
  del.sync(['dist/*', '!dist/webapp']);
  done();
}

gulp.task('clean:war', cleanWar);

function compileNunjucks() {
  return gulp
    .src(['./src/templates/index.nunjucks'])
    .pipe(
      nunjucksRender({
        path: ['./src/templates/']
      })
    )
    .pipe(gulp.dest('./dist'))
    .pipe(devServer.reload());
}

gulp.task('compile:nunjucks', compileNunjucks);

function indexConcatUglify() {
  return gulp
    .src(['./dist/index.html'])
    .pipe(lineEndCorrector({ verbose: true, eolc: 'LF', encoding: 'utf8' }))
    .pipe(useref())
    .pipe(gulpif('*.js', uglify()))
    .pipe(gulpif('*.css', minifyCss({compatibility: 'ie11'})))
    .pipe(gulp.dest('./dist'));
  // console.log('before indexconcat done');
  // done();
}

gulp.task('index:concat:uglify', indexConcatUglify);


function indexMinify() {
  return gulp
  .src(['./dist/index.html'])
  .pipe(htmlmin({ collapseWhitespace: true }))
  .pipe(gulp.dest('./dist'));
}

gulp.task('index:minify', indexMinify);

function cleanDist(done) {
  del.sync([
    './dist/css/**/*',
    '!./dist/css/',
    '!./dist/css/bundle.css',
    './dist/scripts/**/*',
    '!./dist/scripts/',
    '!./dist/scripts/bundle.js',
    '!./dist/scripts/json2.js',
    '!./dist/scripts/rest-ie.js',
    '!./dist/scripts/upgrades',
    '!./dist/scripts/upgrades/**/*'
  ]);
  done();
}

gulp.task('clean:dist', cleanDist);

function compileScss() {
  return gulp
    .src('./src/styles/**/*.scss')
    .pipe(sass().on('error', sass.logError))
    .pipe(
      autoprefixer({
        browsers: ['last 2 versions'],
        cascade: false
      })
    )
    .pipe(gulp.dest('./dist/css'))
    .pipe(devServer.reload());
}

gulp.task('compile:scss', compileScss);

function moveAssets() {
  return gulp.src('./src/assets/**').pipe(gulp.dest('./dist/assets'));
}

gulp.task('move:assets', moveAssets);

function moveImages() {
  return gulp.src(['./src/images/**']).pipe(gulp.dest('./dist/images'));
}

gulp.task('move:images', moveImages);

function moveIcons(done) {
  // return gulp.src('./src/icons/**').pipe(gulp.dest('./dist/css/icons'));
  done();
}

gulp.task('move:icons', moveIcons);

function moveJs() {
  return gulp
    .src(['src/scripts/**/*.js'])
    .pipe(gulp.dest('./dist/scripts'))
    .pipe(devServer.reload());
}

gulp.task('move:js', moveJs);

function moveInf() {
  return gulp.src(['src/INF/**/*', 'src/ping.html']).pipe(gulp.dest('./dist'));
}

gulp.task('move:inf', moveInf);

function moveLang(done) {
  gulp.src(['./src/lang/*']).pipe(gulp.dest('./dist/properties'));
  done();
}

gulp.task('move:lang', moveLang);

function moveConfig() {
  return gulp.src(['./config.properties']).pipe(gulp.dest('./dist'));
}

gulp.task('move:config', moveConfig);

function moveReleaseNotes() {
  return gulp.src(['release-notes/**']).pipe(gulp.dest('dist/release-notes/'));
}

gulp.task('move:release-notes', moveReleaseNotes);

function buildArtifactoryZip(done) {
  try {
    var appData = JSON.parse(fs.readFileSync('./app.json'));
    if (appData) {
      var version = appData.version;
      return gulp
        .src(['dist/**/*'])
        .pipe(zip('counter-' + version + '.zip'))
        .pipe(gulp.dest('dist/'));
    }
  } catch (ex) {
    console.log(
      'There was an exception when trying to read the package.json! - ' + ex
    );
    done(new Error('There was an exception when trying to read the package.json! - ' + ex))
  }
}

gulp.task('build:artifactory:zip', buildArtifactoryZip);

function buildArtifactoryClean(done) {
  del.sync(['dist/*', '!dist/*.zip']);
  done();
}

gulp.task('build:artifactory:clean', buildArtifactoryClean);

function utilWar() {
  return gulp
    .src(['dist/**', '!dist/lang'])
    .pipe(zip('workstationterminal.war'))
    .pipe(gulp.dest('dist/webapp/'));
}

gulp.task('util:war', utilWar);

function watchStart(done) {
  gulp.watch(['./src/styles/**/*.scss'], gulp.parallel('compile:scss'));
  gulp.watch(['src/scripts/**/*.js'], gulp.parallel('move:js'));
  gulp.watch('./src/templates/**/*.nunjucks', gulp.parallel('compile:nunjucks'));
  done();
}

gulp.task('watch:start', watchStart);

function cacheBuster() {
  return gulp
    .src('dist/index.html')
    .pipe(
      cachebust({
        type: 'timestamp'
      })
    )
    .pipe(gulp.dest('dist'));
}

gulp.task('cache:killer', cacheBuster);

function connectDevServer(done) {
  devServer.server({
    root: ['./dist'],
    port: 1337,
    livereload: true,
    middleware: function(connect, opt) {
      return [
        proxy('/rest', {
          target: targetOrchestraUrl,
          route: '/rest',
          changeOrigin: false,
          ws: true
        }),
        proxy('/cometd', {
          target: targetOrchestraUrl,
          route: '/cometd',
          changeOrigin: false,
          ws: true
        }),
        proxy('/bundle', {
          target: targetOrchestraUrl,
          route: '/bundle',
          changeOrigin: true,
          ws: true
        })
      ];
    }
  });
  done();
}

gulp.task('connect', connectDevServer);

function deployWar(done) {
  gulp.src('./dist/webapp/workstationterminal.war').pipe(
    sftp({
      remotePath: remoteDeploymentDefaultPath,
      remotePlatform: remoteDeploymentPlatform,
      host: remoteDeployHost,
      user: remoteDeployUsername,
      pass: remoteDeployPassword
    })
  );
  done();
}

gulp.task('deploy:war', deployWar);

function deployLang(done) {
  gulp
    .src('./dist/properties/workstationTerminalMessages.properties')
    .pipe(
      sftp({
        remotePath: remoteDeploymentDefaultLangPath,
        remotePlatform: remoteDeploymentPlatform,
        host: remoteDeployHost,
        user: remoteDeployUsername,
        pass: remoteDeployPassword
      })
    );
  done();
}

gulp.task('deploy:lang', deployLang);

// Task to create utts files from their source files

function buildUtts() {
  return
}
gulp.task(
  'build:utts',
  folders(uttsPath, function(folder) {
    const currentUttPath = uttsPath + '/' + folder + '/*';
    return gulp
      .src(currentUttPath)
      .pipe(zip(folder + '.utt'))
      .pipe(gulp.dest(path.join(uttsPath, folder)));
  })
);

function cleanUtts(done) {
  del.sync([path.join(uttsPath) + '/**/*.utt']);
  done();
}

gulp.task('clean:utts', cleanUtts);

function moveUtts() {
  return gulp
    .src(uttsPath + '/**/*.utt')
    .pipe(rename({ dirname: '' }))
    .pipe(gulp.dest('./dist/utt'));
}

gulp.task('move:utts', moveUtts);

/**
* Clean and rebuilt Utts
*/
function cleanBuildUtts(done) {
  console.log(`Utt files created from source`);
  done();
}

gulp.task(
  'clean:build:utts',
  gulp.series('clean:utts', 'build:utts', 'move:utts', 'clean:utts')
);

/**
 * Write to manifest file
 */
function writeManifest(done) {
  try {
    var versionInfo = getVersionInfo();
    if (versionInfo) {
      var fileContent = 'Build-Date: ' + new Date().toISOString().substring(0, 10) + '\r\n';
      fileContent += 'Built-By: gulp' + '\r\n';
      fileContent += 'Product-Name: Orchestra Web Counter' + '\r\n';
      fileContent += 'Build-Version: ' + versionInfo.version + '\r\n';
      fs.writeFileSync('src/INF/META-INF/MANIFEST.MF', fileContent);
      done();
    }
  } catch (ex) {
    console.log(
      'There was an exception when trying to read the package.json! - ' + ex
    );
    done(new Error('There was an exception when trying to read the package.json! - ' + ex));
  }
}

gulp.task('write:manifest', writeManifest);

function getVersionInfo() {
  var appData = JSON.parse(fs.readFileSync('./app.json'));
  if (appData) {
    return {
      //versionPrefix: appData.version,
      //version: appData.version + '.' + appData.build,
      //build: appData.build
      version: appData.version
    };
  }
  return null;
}


/**
 * Create customization build, for customization.
 */
function buildCustom(done) {
  console.log(`Build Created in folder ./dist`);
  done();
}

gulp.task(
  'build:custom',
  gulp.series(
    'clean:build',
    'compile:nunjucks',
    'compile:scss',
    'move:js',
    'move:assets',
    'move:images',
    'move:icons',
    'cache:killer',
    'move:inf',
    'clean:build:utts'
  ),
  buildCustom
);




/**
 * Create development build in dist and start watching files for changes
 */

function buildDev(done) {
  console.log(
    `Build Created in folder ./dist - Listening to changes in scripts/styles/templates...`
  );
  done();
}

gulp.task(
  'build:dev',
  gulp.series(
    'clean:build',
    'compile:nunjucks',
    'compile:scss',
    'move:js',
    'move:assets',
    'move:images',
    'move:icons',
    'cache:killer',
    'watch:start',
    'connect'
  ),
  buildDev
);

/**
 * Create developement war
 */

function buildDevWar(done) {
  console.log(
    `workstationterminal.war(Development Build) file created in dist folder`
  );
  done();
}

gulp.task(
  'build:dev:war',
  gulp.series(
    'clean:build',
    'compile:nunjucks',
    'compile:scss',
    'move:js',
    'move:assets',
    'move:images',
    'move:icons',
    'cache:killer',
    'util:war',
    'clean:war',
    'move:lang',
    'clean:build:utts'
  ),
  buildDevWar
);

/**
 * Create developement war
 */

function deployWarArtifactory(done) {
  var warName = fs.readdirSync('./dist')[0];
  var fileExtension = warName.substring(warName.lastIndexOf('.') + 1);
  if (fileExtension === 'zip') {
    ncmd.get(
      `curl -u '${targetArtifactoryUsername}:${targetArtifactoryPassword}' -X PUT ${targetArtifactoryUrl}${targetArtifactoryPath}/${warName} -T ./dist/${warName}`,
      function(err, data, stderr) {
        if (!err) {
          console.log(data);
          done();
        } else {
          console.log(err);
          done(new Error(err));
        }
      }
    );
  } else {
    console.log('Zip file not found!!');
    done(new Error('Zip file not found!!'));
  }
}

gulp.task('deploy:war:artifactory', deployWarArtifactory);

/**
* Create Production war
*/

function buildProdWar(done) {
  console.log(
    `workstationterminal.war(Productiion Build) file created in dist folder`
  );
  done();
}

gulp.task(
  'build:prod:war',
  gulp.series(
    'clean:build',
    'compile:scss',
    'move:js',
    'compile:nunjucks',
    'index:concat:uglify',
    'index:minify',
    'clean:dist',
    'move:assets',
    'move:images',
    'move:icons',
    'cache:killer',
    'move:inf',
    'util:war',
    'clean:war',
    'move:lang',
    'clean:build:utts'
  ),
  buildProdWar
);

/**
* Artifactory build
*/

function buildArtifactory(done) {
  console.log(
    `workstationterminal.war(Productiion Build) file created in dist folder`
  );
  done();
}

gulp.task(
  'build:artifactory',
  gulp.series(
    'clean:build',
    'write:manifest',
    'compile:scss',
    'move:js',
    'compile:nunjucks',
    'index:concat:uglify',
    'index:minify',
    'clean:dist',
    'move:assets',
    'move:images',
    'move:icons',
    'cache:killer',
    'move:inf',
    'util:war',
    'clean:war',
    'move:lang',
    'clean:build:utts',
    'move:release-notes',
    'build:artifactory:zip',
    'build:artifactory:clean'
  ),
  buildArtifactory
);


/**
 *  Deploy war and lang file to a remote Orchestra System
 *  Note - For this to work you need to have openssh installed on the remote server.
 */
function deployRemote(done) {
  console.log(`workstationterminal.war deployed!`);
  done();
}

gulp.task(
  'deploy:remote',
  gulp.series(
    'build:dev',
    'move:inf',
    'util:war',
    'clean:war',
    'move:lang',
    'deploy:war',
    'deploy:lang'
  ),
  deployRemote
);

/**
 *  Deploy war and lang file to a remote Orchestra System
 *  Note - For this to work you need to have openssh installed on the remote server.
 */

function buildArtifactoryDeploy(done) {
  console.log(
    `workstationterminal.zip deployed to artifactory server!`
  );
  done();
}

gulp.task(
  'build:artifactory:deploy',
  gulp.series('build:artifactory', 'deploy:war:artifactory'),
  buildArtifactoryDeploy
);

